yjit/src/codegen.rs

```diff
@@ -2504,18 +2504,18 @@ fn gen_opt_eq(jit: &mut JITState, ctx: &mut Context, cb: &mut CodeBlock, ocb: &m
     }
 }
 
-/*
 fn gen_opt_neq(jit: &mut JITState, ctx: &mut Context, cb: &mut CodeBlock, ocb: &mut OutlinedCb) -> CodegenStatus
 {
     // opt_neq is passed two rb_call_data as arguments:
     // first for ==, second for !=
-    struct rb_call_data *cd = (struct rb_call_data *)jit_get_arg(jit, 1);
-    return gen_send_general(jit, ctx, cd, NULL);
+    let cd = jit_get_arg(jit, 1).as_u64() as *const rb_call_data;
```

Comment:
Hmm, we might want to have an .as_ptr() method to avoid casting as an integer.

Reply:
Yes I'd like to avoid passing by an integer cast, cast from *const u8 to another pointer type.

Can the typename operator and type inference work for us here? Could as_ptr take a pointer type as argument? 🤔

Reply:
It could if we make it a generic, I think, like From. So then we could save the typecast by explicitly adding a type to the "let." It's the same amount of adding types and a bit more complexity in implementation, but arguably more Rust-ified?

Reply:
Yeah could be nice :)

Reply:
Still not completely clear to me what are the limits of Rust's type inference but this is pretty neat :)